File: c/day01.c
#include <stdio.h>

int part1(int m)
{
	return m / 3 - 2;
}

int part2(int m)
{
	int sum = 0;

	while((m = part1(m)) > 0) {
		sum += m;
	}
	return sum;
}

int main(int argc, char **argv)
{
	int m, total = 0, total_fuel = 0;

	while(scanf("%d\n", &m) != EOF) {
		total += part1(m);
		total_fuel += part2(m);
	}

	printf("part 1: %d\n", total);
	printf("part 2: %d\n", total_fuel);

	return 0;
}

File: go/day01.go
package main

import "fmt"

func part1(m int) int {
    return m / 3 - 2
}

func part2(m int) int {
    sum := 0
    for {
        m = part1(m)
        if m <= 0 {
            break
        }
        sum += m
    }
    return sum
}

func main() {
    var m int

    total, total_fuel := 0, 0

    for {
        n, err := fmt.Scanf("%d\n", &m)
        if n == 0 || err != nil {
            break
        }
        total += part1(m)
        total_fuel += part2(m)
    }
	fmt.Printf("part1: %d\n", total)
	fmt.Printf("part2: %d\n", total_fuel)
}

